endpoint/endpoint.go

```diff
@@ -396,3 +397,41 @@ func RemoveDuplicates(endpoints []*Endpoint) []*Endpoint {
 
 	return result
 }
+
+// Check endpoint if is it properly formatted according to RFC standards
+func CheckEndpoint(endpoint Endpoint) bool {
```

Comment:
Can we convert this to a method of the Endpoint type?

func (e *Endpoint) CheckEndpointTargets() bool {
    // function body
}